Neos.ContentRepository.NodeAccess/Classes/FlowQueryOperations/SortOperation.php

```diff
@@ -113,15 +111,12 @@ public function evaluate(FlowQuery $flowQuery, array $arguments)
 
         // Determine the property value to sort by
         foreach ($nodes as $node) {
-            if ($sortProperty[0] === '_') {
-                $propertyValue = \Neos\Utility\ObjectAccess::getPropertyPath($node, substr($sortProperty, 1));
-            } else {
-                $propertyValue = $node->getProperty($sortProperty);
-            }
-
-            if ($propertyValue instanceof \DateTime) {
-                $propertyValue = $propertyValue->getTimestamp();
-            }
+            $propertyValue = match($sortProperty) {
+                '_creationDateTime' => $node->timestamps->created->getTimestamp(),
+                '_lastModificationDateTime' => $node->timestamps->lastModified?->getTimestamp(),
+                '_lastPublicationDateTime' => $node->timestamps->originalLastModified?->getTimestamp(),
+                default => $node->getProperty($sortProperty)
+            };
```

Comment:
I don't like this. If we wouldn't have this syntax in Neos <9, we would never come up with somethink like that.

Why not dedicated operations? sortByCreationDateTime, sortByLastModificationDateTime? 
Or at least sortByTimeStamps(created|lastModified|originalLastModified)?

Reply:
I like this idea!

Reply:
well i would not oppose to that if this is what we agree on thats fine. I just know that christian also agreed with me on doing it like this

> As discussed briefly on the 26.04 we concluded to still allow sorting nodes by date via the `q().sort()` we will likely have to make one exception regarding underscore properties: The sort operation will translate `_lastModificationDateTime` and respective things to the 9.0 equivalent logic internally instead of having to introduce another  sort operation or a special parameter or another special syntax like `.sort(['$$lastModified'])` or such.

so just let me know if we should go down dennys proposed route then ill implement it.

Reply:
introduced `sortByTimestamp` fq